### PR TITLE
Rename iulia to yul in documentation.

### LINKED ISSUE #3791: Update documentation: change Julia to yul

### PR DESCRIPTION
**Pull request by chriseth**
We should also change the commandline interface (in a separate PR).

Fixes #3791. Part of #3241.

### CODE REVIEW

File: docs/yul.rst [hunk 1]
```diff
@@ -69,10 +70,10 @@ and ``add`` to be available.
         }
     }
 
-Specification of JULIA
-======================
+Specification of Yur
```
Comment (ekpyron) [inline docs/yul.rst]: Typo ``Yur``

**Review by ekpyron — DISMISSED** [review]

File: docs/yul.rst [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-Joyfully Universal Language for (Inline) Assembly
-#################################################
+###
+Yul
+###
 
 .. _julia:
 
-.. index:: ! assembly, ! asm, ! evmasm, ! julia
+.. index:: ! assembly, ! asm, ! evmasm, ! yul, julia
```
Comment (aj07) [inline docs/yul.rst]: IMO julia should be removed

File: docs/yul.rst [hunk 3]
```diff
@@ -1,18 +1,19 @@
-#################################################
-Joyfully Universal Language for (Inline) Assembly
-#################################################
+###
+Yul
+###
 
 .. _julia:
```
Comment (axic) [inline docs/yul.rst]: Where is this section name used? Can we rename it now?

File: docs/yul.rst (hunk 2 above)
Comment (axic) [inline docs/yul.rst]: If we're keeping `julia` as search keyword, maybe keep `iulia` too? Otherwise I'd remove.

File: docs/yul.rst (hunk 3 above)
Comment (axic) [reply docs/yul.rst]: Just checked, this doesn't seem to be referred to yet so should be safe to rename.

**Review by axic — DISMISSED** [review]

**Review by axic — APPROVED** [review]

**Pull request merged**